packages/next/src/server/app-render/use-flight-response.tsx

```diff
@@ -78,6 +79,25 @@ export async function flightRenderComplete(
   }
 }
 
+const decoder = new TextDecoder('utf-8', { fatal: true })
```

Comment:
This is not safe because there can be multiple streams at the same time interleaving and because the streaming mode is used, there's a buffer stored inside this object for partial bytes received from previous chunks which can conflict.

This needs to be a new one per stream like it was before.

Reply:
Makes total sense!